src/domain/entities/SynchronizationResult.ts

```diff
@@ -12,6 +12,7 @@ export interface SynchronizationStats {
 export interface ErrorMessage {
     id: string;
     message: string;
+    details?: string;
```

Comment:
As a rule of thumb, use as few `optional?` fields as possible. This way the caller is then forced to set a value and cannot ignore it. Typically: `detail: string | undefined`